src/core/src/package_managers/YumPackageManager.py

```diff
@@ -282,7 +287,121 @@ def is_package_version_installed(self, package_name, package_version):
 
         return False
 
-    def extract_dependencies(self, output, packages):
+    def extract_dependencies_yum_4_or_higher(self, output, packages):
+        # Sample 1 (Updates are available and there is no new package installation required):
+        # Sample output for the cmd 'sudo yum update --assumeno selinux-policy.noarch' is :
+        #
+        #Last metadata expiration check: 0:08:56 ago on Tue 25 Jul 2023 02:14:28 PM UTC.
+        #Package selinux-policy-3.14.3-95.el8_6.6.noarch is already installed.
+        #Dependencies resolved.
+        #==================================================================================================
+        # Package                   Arch    Version           Repository                               Size
+        #==================================================================================================
+        #Upgrading:
+        # selinux-policy            noarch  3.14.3-95.el8_6.8 rhel-8-for-x86_64-baseos-eus-rhui-rpms  651 k
+        # selinux-policy-targeted   noarch  3.14.3-95.el8_6.8 rhel-8-for-x86_64-baseos-eus-rhui-rpms   15 M
+        #
+        #Transaction Summary
+        #==================================================================================================
+        #Upgrade  2 Packages
+        #
+        #Total download size: 16 M
+        #Operation aborted..
+        
+        # Sample 2 (There are no new updates but there are new package installations):
+        # Sample output for the cmd 'sudo yum update --assumeno kernel-modules.x86_64' is :
+        #
+        #Last metadata expiration check: 0:09:14 ago on Tue 25 Jul 2023 02:14:28 PM UTC.
+        #Package kernel-modules-4.18.0-372.9.1.el8.x86_64 is already installed.
+        #Package kernel-modules-4.18.0-372.52.1.el8_6.x86_64 is already installed.
+        #Dependencies resolved.
+        #=============================================================================================
+        # Package          Arch    Version               Repository                               Size
+        #=============================================================================================
+        #Installing dependencies:
+        # kernel-core      x86_64  4.18.0-372.64.1.el8_6 rhel-8-for-x86_64-baseos-eus-rhui-rpms   40 M
+        # kernel-modules   x86_64  4.18.0-372.64.1.el8_6 rhel-8-for-x86_64-baseos-eus-rhui-rpms   32 M
+        #
+        #Transaction Summary
+        #=============================================================================================
+        #Install  2 Packages
+        #
+        #Total download size: 72 M
+        #Installed size: 93 M
+        #Operation aborted.
+        
+        # In sample1, the starting delimeter is "Upgrading:" and in sample2, the starting delimeter is "Installing dependencies:"
+        # If there are combinations of updates and new installations, one of the above delimeters will come first and the below logic will work.
```

Comment:
Please list a sample of what is seen when there are combinations of updates and installations.

Something to pay special attention to here is, where does "Transaction summary" appear in this instance. The implementation below breaks out of the for loop upon encountering "Transaction summary". In a case where there are both upgrades and installations and 2 separate transaction summaries (one for each), the loop would stop after the first one and never parse the 2nd list

Reply:
|	Last metadata expiration check: 0:01:56 ago on Tue 25 Jul 2023 12:01:47 PM UTC.
	|	Dependencies resolved.
	|	================================================================================================
	|	 Package             Arch    Version               Repository                               Size
	|	================================================================================================
	|	Upgrading:
	|	 kernel-tools        x86_64  4.18.0-372.64.1.el8_6 rhel-8-for-x86_64-baseos-eus-rhui-rpms  8.4 M
	|	 kernel-tools-libs   x86_64  4.18.0-372.64.1.el8_6 rhel-8-for-x86_64-baseos-eus-rhui-rpms  8.2 M
	|	 openssl             x86_64  1:1.1.1k-9.el8_6      rhel-8-for-x86_64-baseos-eus-rhui-rpms  710 k
	|	 openssl-libs        x86_64  1:1.1.1k-9.el8_6      rhel-8-for-x86_64-baseos-eus-rhui-rpms  1.5 M
	|	Installing dependencies:
	|	 kernel-core         x86_64  4.18.0-372.64.1.el8_6 rhel-8-for-x86_64-baseos-eus-rhui-rpms   40 M
	|	 kernel-modules      x86_64  4.18.0-372.64.1.el8_6 rhel-8-for-x86_64-baseos-eus-rhui-rpms   32 M
	|	
	|	Transaction Summary
	|	================================================================================================
	|	Install  2 Packages
	|	Upgrade  4 Packages

Reply:
Above is the sample of what is seen when there are combinations of updates and installations.
The string "Transaction Summary" comes after both sections "Upgrading" and "Installing dependencies" are completed.

Should I add this sample in the code as a comment? I did not add it because there is already long comment in this function. Might be we can just mention in a comment that "Transaction Summary" comes after both sections of updates and installations are completed.

Reply:
Better to keep it in code as a comment, will act as a reference for future

Reply:
Added the sample output as comment